packages/app/src/components/DBSessionPanel.tsx

```diff
@@ -0,0 +1,139 @@
+import { useMemo } from 'react';
+import { TSource } from '@hyperdx/common-utils/dist/types';
+import { Loader } from '@mantine/core';
+
+import SessionSubpanel from '@/SessionSubpanel';
+import { useSource } from '@/source';
+import { getDisplayedTimestampValueExpression } from '@/source';
+
+import { useEventsData } from './DBTraceWaterfallChart';
+
+export const useSessionId = ({
+  sourceId,
+  traceId,
+  dateRange,
+  enabled = false,
+}: {
+  sourceId?: string;
+  traceId: string;
+  dateRange: [Date, Date];
+  enabled?: boolean;
+}) => {
+  const { data: source } = useSource({ id: sourceId });
+
+  const config = useMemo(() => {
+    if (!source) {
+      return;
+    }
+    return {
+      select: [
+        {
+          valueExpression: getDisplayedTimestampValueExpression(source),
+          alias: 'Timestamp',
+        },
+        {
+          valueExpression: `${source.resourceAttributesExpression}['rum.sessionId']`,
+          alias: 'rumSessionId',
+        },
+        {
+          valueExpression: `${source.resourceAttributesExpression}['service.name']`,
+          alias: 'serviceName',
+        },
+        {
+          valueExpression: `${source.parentSpanIdExpression}`,
+          alias: 'parentSpanId',
+        },
+      ],
+      from: source.from,
+      timestampValueExpression: getDisplayedTimestampValueExpression(source),
+      limit: { limit: 10000 },
+      connection: source.connection,
+      where: `${source.traceIdExpression} = '${traceId}'`,
+      whereLanguage: 'sql' as const,
+    };
+  }, [source, traceId]);
+
+  const { data } = useEventsData({
```

Comment:
style: `useEventsData` seems to be used for `DBTraceWaterfallChartContainer` particularly. I think here we can just make a call using `clickhouseClient.query` like https://github.com/hyperdxio/hyperdx/blob/b66fc1464166dffaf06258f9c77094f3f9f3dd85/packages/app/src/clickhouse.ts#L244-L249